lms/services/application_instance.py

```diff
@@ -1,34 +1,41 @@
-from typing import Optional
+from functools import lru_cache
 
 from lms.models import ApplicationInstance
+from lms.services import ConsumerKeyError
 
 
 class ApplicationInstanceService:
-    def __init__(self, db):
+    def __init__(self, db, default_consumer_key):
         self._db = db
+        self._default_consumer_key = default_consumer_key
 
-    def get(self, consumer_key: str) -> Optional[ApplicationInstance]:
-        """Query one ApplicationInstance by consumer_key."""
+    @lru_cache
```

Comment:
What's the default size on `lru_cache`?

Reply:
I wonder if this is a case where having the service actually be persistent and cache across requests would be good, or terrible and confusing?

We'd need to actually do the interface thing though